### ISSUE TITLE
Adding second event listener breaks scroll reveal

### ISSUE THREAD

**Issue by coopersamuel** [jlmakes/scrollreveal#437]
Hey Julian,

I'm using ScrollReveal and have it working great, until I add a second scroll event listener:
`window.addEventListener('scroll', (event) => { ... });`

It looks like this new event listener is breaking ScrollReveal somehow. This is the error message I'm getting:
```
Uncaught TypeError: Cannot read property 'bounds' of undefined
    at ScrollReveal.isElementVisible (scrollreveal.js:1340)
    at scrollreveal.js:1398
    at scrollreveal.js:181
```

**Comment by jlmakes** [comment]
Hey Samuel,

This error message is similar to #349, but I believe it was due to element/container data being destroyed during animation (such as calling `clean()` midway through a reveal sequence). I concluded this was exceedingly uncommon, and put the issue on ice.

When I added `window.addEventListener('scroll', () => console.log("whammy"));` to a local test environment, I didn't run into any issues.

Would you mind recreating this with a sandbox like [JSBin](http://jsbin.com)?

**Comment by coopersamuel** [comment]
Hey Julian, 

I can't seem to recreate the issue either... sorry about this, but maybe it is an uncommon issue.
I found a workaround for my project, so I guess this can go back on ice...

~ Sam

**Comment by jlmakes** [comment]
No worries, glad you found a workaround!

**Comment by jorgeemherrera** [comment]
Hi jlmakes, I have the same issue, the first time the opacity element of list is showed, but if you go to other link and come back, the las section put the opacity:0  and the error:

> ERROR TypeError: Cannot read property 'bounds' of undefined
    at ScrollReveal.isElementVisible (scrollreveal.js:1363)
    at scrollreveal.js:1421
    at scrollreveal.js:199
    at Array.forEach (<anonymous>)
    at each (scrollreveal.js:199)
    at scrollreveal.js:1417
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:423)
    at Object.onInvokeTask (core.js:17290)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:422)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:195)

**Comment by mikehoh** [comment]
The same problem in vuejs app (SPA mode). 
- Open page, block appears with sr.
- For example we have a link inside this block
- Click on it and it opens another page
- Go back to previous page
- Get this error in console:
```
Uncaught TypeError: Cannot read property 'bounds' of undefined
    at ScrollReveal.isElementVisible (scrollreveal.es.js?6e55:989)
    at eval (scrollreveal.es.js?6e55:1047)
    at eval (scrollreveal.es.js?6e55:89)
    at Array.forEach (<anonymous>)
    at each (scrollreveal.es.js?6e55:89)
    at eval (scrollreveal.es.js?6e55:1043)
```

**Comment by jlmakes** [comment]
Hmm, perhaps the stateful nature of ScrollReveal is causing problems: it sounds like internal store has a reference to an element without geometry data (i.e. bounds and position).

Hey @mikehoh, just out of curiosity, do you mind making a quick change to your local copy of ScrollReveal to test a possible solution?

https://github.com/scrollreveal/scrollreveal/blob/c98fad79f4750dfd80797414e7d94f1cddd2109c/src/instance/functions/delegate.js#L37-L42

On line `38` here, we should try to expand the conditions:

```diff
- if (stale) {
+ if (stale || !'geometry' in element) {
```

Thanks Mike!

**Comment by jlmakes** [comment]
Hmm, perhaps the stateful nature of ScrollReveal is causing problems: It sounds like internal store has a reference to an element without geometry data (i.e. bounds and position).

Hey @mikehoh, just out of curiosity, do you mind making a quick change to your local copy of ScrollReveal to test a possible solution?

https://github.com/scrollreveal/scrollreveal/blob/c98fad79f4750dfd80797414e7d94f1cddd2109c/src/instance/functions/delegate.js#L37-L42

On line `38` here, we should try to expand the condition:

```diff
- if (stale) {
+ if (stale || element.geometry === undefined) {
```

Thanks Mike!

**Comment by alexwidua** [comment]
Hey @jlmakes, I'm a bit late to the party but I ran into the exact same problem as @mikehoh (using Vuejs SPA, going back and forth in history results in a `Cannot read property 'bounds' of undefined` error) and your proposed solution solves the problem.

**Comment by matthewattanasio** [comment]
Hey @jlmakes and @alexwidua I am using Vuejs with Gridsome so I believe its a SPA. I am using  [Vue-scrollTo](https://www.npmjs.com/package/vue-scrollto) which uses ScrollReveal at the latest version of 4.0.6 for scrollReveal. However, I am still getting the exact same error `Cannot read property 'bounds' of undefined`

Any help would be much appreciated.

**Comment by jlmakes** [comment]
@matthewattanasio I'll publish the fix tomorrow as `4.0.7`

**Comment by matthewattanasio** [comment]
awesome @jlmakes , looking forward to it

**Comment by jlmakes** [comment]
Had a tough Monday, but thanks for the reminder @matthewattanasio! Just pushed `4.0.7`, thanks again @alexwidua!

**Comment by jlmakes** [comment]
Thanks for the reminder @matthewattanasio! Just pushed `4.0.7`, thanks again @alexwidua!

**Comment by jlmakes** [comment]
Thanks for the reminder @matthewattanasio! Just published `4.0.7` 🚀 Thanks again @alexwidua!
